SourceryFramework/Sources/Parsing/FileParser.swift

```diff
@@ -647,6 +647,11 @@ extension FileParser {
             return nil
         }
 
+        var isIndirect: Bool = false
+        if let attributes = source["key.attributes"] as? [[String: SourceKitRepresentable]], let attribute = attributes[0]["key.attribute"] as? String, SwiftDeclarationAttributeKind(rawValue: attribute) == .indirect {
```

Comment:
I have to admit this isn't super clear code, but I couldn't see what the preferred way of digging into these `SourceKitRepresentable`s is so I thought I would call it out here. What might be a cleaner way to extract the attribute from `source`?

Reply:
I think this is fine for this field, but shouldn't you guard against empty array? either use `first` or add a check before indexing

Reply:
Good catch! I've amended to use `.first?` in place of `[0]`.